### PR TITLE
chore: cleanup decoder benches

### PR DESCRIPTION
**Pull request by westonpace**
This ensures that the setup routines happen inside of the benchmark function.  Otherwise datagen will be run for all benchmarks during the collection phase and the system will run out of RAM.

This also changes the routines to use `FileMetadataCache::no_cache` as it turns out `moka` doesn't immediately free its cache space on drop and, when used in a tight benchmark loop, this leads to running out of memory.

### CODE REVIEW

File: rust/lance-encoding/benches/decoder.rs [hunk 1]
```diff
@@ -201,114 +227,125 @@ fn bench_decode_packed_struct(c: &mut Criterion) {
...
+            "string",
+            DataType::Utf8,
+            false,
+        )]));
+
+        let data = RecordBatch::try_new(schema, string_data.columns().to_vec()).unwrap();
+
+        let lance_schema =
+            Arc::new(lance_core::datatypes::Schema::try_from(data.schema().as_ref()).unwrap());
+        let encoding_strategy = default_encoding_strategy(LanceFileVersion::default());
```
Comment (broccoliSpicy) [inline rust/lance-encoding/benches/decoder.rs:331]: I think fixed size binary encoding is not enabled in `LanceFileVersion::default`(Lance 2.0)

File: rust/lance-encoding/benches/decoder.rs [hunk 2]
```diff
@@ -65,33 +55,37 @@ const ENCODING_OPTIONS: EncodingOptions = EncodingOptions {
...
         let func_name = format!("{:?}", data_type).to_lowercase();
+        let num_rows = NUM_BYTES / data_type.primitive_width().unwrap() as u64;
         group.bench_function(func_name, |b| {
+            let data = lance_datagen::gen()
+                .anon_col(lance_datagen::array::rand_type(data_type))
+                .into_batch_rows(lance_datagen::RowCount::from(num_rows))
+                .unwrap();
+            let lance_schema =
+                Arc::new(lance_core::datatypes::Schema::try_from(data.schema().as_ref()).unwrap());
+            let encoding_strategy = default_encoding_strategy(LanceFileVersion::default());
```
Comment (broccoliSpicy) [inline rust/lance-encoding/benches/decoder.rs:70]: is this intentional that we only run `bench_decode` against 2.0(LanceFileVersion::default)?

**Comment by broccoliSpicy** [comment]
related to #2725

**Review by broccoliSpicy — APPROVED** [review]

File: rust/lance-encoding/benches/decoder.rs (hunk 1 above)
Comment (westonpace) [reply rust/lance-encoding/benches/decoder.rs:332]: I'm not sure but I didn't change the benchmark.  We can revisit when we test fixed-size-binary in 2.1

File: rust/lance-encoding/benches/decoder.rs (hunk 2 above)
Comment (westonpace) [reply rust/lance-encoding/benches/decoder.rs:70]: For now, yes.  I think I updated one or two benchmarks to run against 2.1 but wasn't trying to update the whole suite yet.

**Pull request merged**